datashader/utils.py

```diff
@@ -154,6 +166,73 @@ def get_indices(x, y, xs, ys, res):
     return int(x_), int(y_)
 
 
+def orient_array(raster, res, band):
+    """
+    Reorients the array to a canonical orientation depending on
+    whether the x and y-resolution values are positive or negative.
+
+    Parameters
+    ----------
+    raster : DataArray
+        xarray DataArray to be reoriented
+    res : tuple
+        Two-tuple (int, int) which includes x and y resolutions (aka "grid/cell
+        sizes"), respectively.
+    band : int
+        Index of the raster band to be reoriented (optional)
+
+    Returns
+    -------
+    array : numpy.ndarray
+        Reoriented 2d NumPy ndarray
+    """
+    array = raster.data
+    if array.ndim == 2:
+        if res[0] < 0: array = array[:, ::-1]
+        if res[1] > 0: array = array[::-1]
+    else:
+        if band is not None: array = array[band-1]
+        if res[0] < 0: array = array[:, :, ::-1]
+        if res[1] > 0: array = array[:, ::-1]
+    return array
+
+
+def compute_coords(width, height, x_range, y_range, res):
```

Comment:
Is `compute_coords` still necessary, since we are now relying on xarray to provide bin-centered coordinates?